lib/ts/recipe/passwordless/components/features/linkClickedScreen/index.tsx

```diff
@@ -38,15 +38,22 @@ type PropType = FeatureBaseProps & { recipe: Recipe };
 
 const LinkClickedScreen: React.FC<PropType> = (props) => {
     const userContext = useUserContext();
+    const [showButton, setRequiresInteraction] = useState<boolean>(false);
 
     const consumeCode = useCallback(async () => {
         const preAuthSessionId = getQueryParams("preAuthSessionId");
         const linkCode = getURLHash();
 
         if (preAuthSessionId === null || preAuthSessionId.length === 0 || linkCode.length === 0) {
-            return props.recipe.redirectToAuthWithoutRedirectToPath(undefined, props.history, {
+            await props.recipe.redirectToAuthWithoutRedirectToPath(undefined, props.history, {
                 error: "signin",
             });
+            return "REDIRECTING";
+        }
+        const loginAttemptInfo = await props.recipe.recipeImpl.getLoginAttemptInfo({ userContext });
+
+        if (loginAttemptInfo?.preAuthSessionId !== preAuthSessionId) {
```

Comment:
if `loginAttemptInfo?.preAuthSessionId === undefined`, then it requires interaction. On the other hand, if `loginAttemptInfo?.preAuthSessionId !== preAuthSessionId`, then don't you want to show `error: "signin"`?

Reply:
Not necessarily. Maybe we could add an additional warning to this page, but I don't think we should redirect since logging in through the link is not affected by the stored info. It could indicate, for example, a resent link or some outdated info stored in the current browser if the user wants to continue here after starting the login process elsewhere.

Reply:
Can we discuss this in more details: What are all the cases that this if statement would be true in?

Reply:
Two groups of cases:
1. Continuing on a new device (w/ no stored info)
2. Stored info about a different attempt

A different attempt could be:
1. Out of date attempt (someone started a login attempt in the same browser)
2. Current attempt from the same user, e.g.: they hit resend but clicked the first email they received, not the last.
3. Current attempt from a diff user - one account started a login process then a link for a different account is clicked

I think in all cases, we are logging in to the expected user and behaving exactly as expected.

Reply:
Ok agreed.